sycl/include/sycl/ext/intel/experimental/esimd/common.hpp

```diff
@@ -50,57 +58,162 @@ using uint = unsigned int;
 // functions defined in a header.
 #define ESIMD_INLINE inline __attribute__((always_inline))
 
-// Enums
-// TODO FIXME convert the two enums below to nested enum or class enum to
-// remove enum values from the global namespace
-enum { GENX_NOSAT = 0, GENX_SAT };
-
-enum ChannelMaskType {
-  ESIMD_R_ENABLE = 1,
-  ESIMD_G_ENABLE = 2,
-  ESIMD_GR_ENABLE = 3,
-  ESIMD_B_ENABLE = 4,
-  ESIMD_BR_ENABLE = 5,
-  ESIMD_BG_ENABLE = 6,
-  ESIMD_BGR_ENABLE = 7,
-  ESIMD_A_ENABLE = 8,
-  ESIMD_AR_ENABLE = 9,
-  ESIMD_AG_ENABLE = 10,
-  ESIMD_AGR_ENABLE = 11,
-  ESIMD_AB_ENABLE = 12,
-  ESIMD_ABR_ENABLE = 13,
-  ESIMD_ABG_ENABLE = 14,
-  ESIMD_ABGR_ENABLE = 15
+// Macros for internal use
+#define __ESIMD_NS sycl::ext::intel::experimental::esimd
+#define __ESIMD_QUOTE1(m) #m
+#define __ESIMD_QUOTE(m) __ESIMD_QUOTE1(m)
+#define __ESIMD_NS_QUOTED __ESIMD_QUOTE(__ESIMD_NS)
+#define __ESIMD_DEPRECATED(new_api)                                            \
```

Comment:
nit: should we undef those defines as well.

Reply:
I think those can be useful in other headers.
__ESIMD_NS should actually replace __SEIEE according to prior reviews.